client/components/App.jsx

```diff
@@ -1,18 +1,50 @@
 import React, { Component } from 'react';
+import axios from 'axios';
+
+import Header from './Header';
+import RatingsList from './RatingsList';
+import ReviewsList from './ReviewsList';
+import Modal from './Modal';
+import ModalButton from './ModalButton';
 
 class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      message: 'Howdy',
+      rating: {},
+      reviews: [],
+      modalView: false,
     };
+    this.switchModal = this.switchModal.bind(this);
+  }
+
+  async componentDidMount() {
+    try {
+      const res = await axios.get('/api/reviews/0/');
+      this.setState(res.data);
+    } catch (error) { console.log(error); }
+  }
+
+  switchModal(e) {
+    const { modalView } = this.state;
+    this.setState({
+      modalView: !modalView,
+    });
   }
 
   render() {
-    const { message } = this.state;
+    const { rating, reviews } = this.state;
+    const modalButtonText = `Show all ${reviews.length} reviews`;
     return (
-      <h1>{message}</h1>
+      <div id="reviewsComponent">
+        <Header rating={rating.overall} numOfReviews={reviews.length} />
+        <RatingsList rating={rating} />
+        <ReviewsList reviews={reviews} />
+        <ModalButton switchModal={this.switchModal} text={modalButtonText} />
+        <Modal mainState={this.state} switchModal={this.switchModal} />
```

Comment:
I think it'd be a good idea to move the reviews state into it's own object, like 
`this.state = {
   reviewInfo: {...}
}`
It's sort of an anti-pattern to pass all of the state the child component, as if they should have more equal or more context than the parent.
Imagine the case where you add a few more things to the App state, then you're going to have to change this line anyways.

Reply:
On that note,
it might even be a better idea to conditionally render the Modal component in the parent container, like so
```suggestion
        {
           modalView
              ? <Modal mainState={this.state} switchModal={this.switchModal} />
              : null;
        }
```